numba/tests/npyufunc/test_gufunc.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import absolute_import, print_function, division
+from numba import unittest_support as unittest
+from numba.decorators import jit
+from numba import float32
+import numpy as np
+import numpy.core.umath_tests as ut
+from numba.npyufunc import GUVectorize
+
+
+def matmulcore(A, B, C):
+    m, n = A.shape
+    n, p = B.shape
+    for i in range(m):
+        for j in range(p):
+            C[i, j] = 0
+            for k in range(n):
+                C[i, j] += A[i, k] * B[k, j]
+
+
+class TestGUFunc(unittest.TestCase):
```

Comment:
I had planned to merge test_ufuncbuilding.py and test_guvectorize_scalar.py into the new tests later, but I can certainly do that right now. I'm not sure about duplication, as I don't (yet) fully understand all that is being tested there, so rather than simply removing tests I'd just keep them closely together for us to determine redundancy later on.